run-tests.php

```diff
@@ -2971,24 +2972,31 @@ function settings2params(array $ini_settings): string
     $settings = '';
 
     foreach ($ini_settings as $name => $value) {
-        if (is_array($value)) {
-            foreach ($value as $val) {
-                $val = addslashes($val);
-                $settings .= " -d \"$name=$val\"";
-            }
-        } else {
-            if (IS_WINDOWS && !empty($value) && $value[0] == '"') {
-                $len = strlen($value);
-
-                if ($value[$len - 1] == '"') {
-                    $value[0] = "'";
-                    $value[$len - 1] = "'";
+        if (IS_WINDOWS) {
+            if (is_array($value)) {
+                foreach ($value as $val) {
+                    $val = addslashes($val);
```

Comment:
I don't know enough about how windows command-line-arguments are constructed, to know if this should really be addslashes() or escapeshellarg() (i'm guessing the latter but i'm not sure)

Wouldn't surprise me if the answer lies in here though: https://docs.microsoft.com/en-us/archive/blogs/twistylittlepassagesallalike/everyone-quotes-command-line-arguments-the-wrong-way